crates/goose/src/agents/subagent_handler.rs

```diff
@@ -117,10 +117,11 @@ fn get_agent_messages(
     cancellation_token: Option<CancellationToken>,
 ) -> AgentMessagesFuture {
     Box::pin(async move {
-        let text_instruction = recipe
-            .instructions
+        let system_instructions = recipe.instructions.clone().unwrap_or_default();
+        let user_task = recipe
+            .prompt
             .clone()
-            .or(recipe.prompt.clone())
+            .or(recipe.instructions.clone())
```

Comment:
this should probably not also default to instructions, since a recipe might not have a prompt, in which case we should probably just say, ehm, "do it" ?